src/frontend/src/pages/PIAForm/index.tsx

```diff
@@ -329,7 +329,21 @@ const PIAFormPage = () => {
               ? PiaStatuses.MPO_REVIEW
               : PiaStatuses.INCOMPLETE,
         });
-        if (pia?.id) {
+        if (
+          (pia?.id &&
+            pia?.isNextStepsSeenForNonDelegatedFlow === true &&
+            (pia?.hasAddedPiToDataElements === PIOptions[0].value ||
```

Comment:
Agree with your statement. Personally I do not like referencing index by constants. good'' to create a map
const enum {
'YES' //index 0
'NO' // index 1
'NOTSURE' // index 2
}